common/taskvalidator/validateworkflow_test.go

```diff
@@ -25,34 +25,43 @@ package taskvalidator
 import (
 	"testing"
 
+	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/assert"
 
+	"github.com/uber/cadence/common/cache"
 	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/service/history/constants"
 )
 
-// MockMetricsScope implements the metrics.Scope interface for testing purposes.
-type MockMetricsScope struct{}
-
-func (s *MockMetricsScope) IncCounter(counter int) {}
-
 func TestWorkflowCheckforValidation(t *testing.T) {
-	// Create a mock logger and metrics client
-	logger := log.NewNoop()
-	metricsClient := metrics.NewNoopMetricsClient()
+	testCases := []struct {
+		name     string
+		domainID string
+		runID    string
+	}{
+		{"DomainFetchSuccess", "domain-id-success", "run-id-success"},
+		{"DomainFetchFailure", "domain-id-failure", "run-id-failure"},
```

Comment:
how are these 2 cases different? validations expect no error below

Reply:
The main code currently doesn't do anything except for logging the case. The test is for a supporting domain id retrieval check as there's not much to test really. Will be modifying it with something more substantial in the upcoming PR.